### DISCUSSION TITLE
Is there a way to split one event into multiple events according to custom logic?

### DISCUSSION THREAD

**Discussion by zmingz123** [vectordotdev/vector#11796] [Q&A] [answered]
Hi there，
I want to split one event into multiple events according to custom logic, like [Record Split in Fluent-bit](https://docs.fluentbit.io/manual/pipeline/filters/lua#record-split).
for example：
input:
```
{
"a":"11",
"b":"11",
"c":"11",
"d":"11",
"extra":[
{"aa":"1"},
{"bb":"2}
]
}
```
target output:
```
{"a":"11","b":"11","c":"11","d":"11","aa":"1"}
{"a":"11","b":"11","c":"11","d":"11","bb":"2"}
```
I tried `parse_json` but it seems can only handle arrays

**Comment by spencergilbert** [comment, marked as answer]
I think [`unnest`](https://vector.dev/docs/reference/vrl/functions/#unnest) may cover this for you:

```coffee
. = unnest!(.extra)
```
